### ISSUE TITLE
Crash when running on gpu

### ISSUE THREAD

**Issue by MathieuGrosso** [OpenNMT/CTranslate2#780]
Hello,
I am trying to use ctranslate2.translate_batch on gpu but i keep getting the same error:
"RuntimeError: cuBLAS failed with status CUBLAS_STATUS_EXECUTION_FAILED".
I have cuda version :11.6, ctranslate2: 2.15.1, and driver version : 510.47.03.

Thanks a lot, if you know what is going on, it would really help me.
Mathieu

**Comment by guillaumekln** [comment]
Hi,

What is the GPU model?

Can you also post a code snippet that raises this error?

**Comment by MathieuGrosso** [comment]
Hello,
Thanks for reply:
Gpu is Tesla V100-PCIE (16go ram)
The error is:
"Output=translator.translate_batch(
RuntimeError: cuBLAS failed with status CUBLAS_STATUS_EXECUTION_FAILED"

It works fine on cpu by the way.
Thanks !

**Comment by MathieuGrosso** [comment]
Yes sorry, 
`
Device ="cuda"
Translator = ctranslate2.Translator(model_path=model_path,device=device,device_index=0)
Sp=spm.sentenpieceprocessor(model_file=model_file)
With open(src, 'r', encoding='utf8') as f: 
      x=f.read().split('\n')
x_spm = sp.encode(x,out_type='str') 
x_spm = [[f'__{input_language}__"] for _ in range(len(x_spm))], max_batch_size=1, beam_size=5)

Res=[]
For o in output : 
       res.append(sp.decode(o.hypotheses[0][1:]))

With open(out,'w') as f:
      f.write('\n'.join(res)) 
`

Sorry if there are some typos. I am not allowed to copy/paste from my computer so I had to rewrite everything on my phone.

Let me know if thats helpful. Thanks again.

**Comment by guillaumekln** [comment]
I'm not reproducing this error on a V100S with the same driver and CUDA versions.

Can you enable the info logs and report the output? Set the environment variable `CT2_VERBOSE=1`, e.g.:

```text
CT2_VERBOSE=1 python3 ...
```

**Comment by MathieuGrosso** [comment]
![20220425_150431](https://user-images.githubusercontent.com/76784995/165095955-91c67101-9847-4aa6-96cc-f55a2ee0d9f4.jpg)
Hello, here are the logs.
Thanks again

**Comment by guillaumekln** [comment]
> Loaded cuBLAS library version 10.2.1

This is unexpected. The code is searching for `libcublas.so.11` from CUDA 11 and somehow your system returns `libcublas.so.10` from CUDA 10 instead. Something looks wrong with your CUDA installation.

What is the output of the command below?

```bash
ldconfig -p | grep cublas
```

**Comment by MathieuGrosso** [comment]
Yes ! That's what I thought too, looks like a problem of version.
When using this command I get this: 
![20220425_155334](https://user-images.githubusercontent.com/76784995/165103594-d27828f0-0807-4e29-b0f4-6fca2b438ed2.jpg)

Tell me if that's helpful

**Comment by guillaumekln** [comment]
Can you follow the links to the end?

```bash
ls -l /usr/local/cuda/targets/x86_64-linux/lib/libcublas.so*
```

**Comment by MathieuGrosso** [comment]
Yes i have it when using this command: 
![20220425_160952](https://user-images.githubusercontent.com/76784995/165106536-4ddeabe5-0ea8-4993-b150-f445b0ec3f04.jpg)

I found it surprising since this version seem to be 11.X...

**Comment by guillaumekln** [comment]
Yes, it's not clear what the issue is based on these outputs. You should find where this older cuBLAS version is coming from and either remove it or fix the symbolic links.

**Comment by guillaumekln** [comment]
Did you find the reason for this incorrect cuBLAS version?

I'm going to close this issue since it is an installation/environment issue and not something we can fix in the code.
